packages/git/src/browser/dirty-diff/dirty-diff-manager.ts

```diff
@@ -136,6 +139,18 @@ export class DirtyDiffManager {
         return undefined;
     }
 
+    protected async canHandle(uri: string): Promise<boolean> {
+        if (new URI(uri).scheme !== 'file') {
+            return false;
+        }
+        const repository = this.repositoryTracker.selectedRepository;
```

Comment:
I'm not sure how achieve this, but could we open a file in the editor, and the `selectedRepository` will be first `undefined`, but if this changes afterwards, I'd like to see the dirty diffs.

Reply:
I can understand your concern about this "enhancement", but that is unrelated to the `fatal ... is outside repository` issue this PR is intended to solve. Or did I miss something?

Reply:
Yes, I tried to make it clearer in the other comment. The manager should not exclude file URIs if the repository is not available, or the file is not version controlled, as it might change later on, right?